### ISSUE TITLE
[Feature]: Discussion - Use tags in replacement (or additionally to) albums

### ISSUE THREAD

**Issue by Tonux599** [immich-app/immich#838]
### Feature detail

I personally would be keen to see support for tagging photos. I like how digiKam does this, such that any single piece of media can have any number of tags applied to it.

How albums are currently implemented is close to this. Such that a single piece of media can belong to multiple albums. Notably however, what is absent, is no way of viewing media that is not assigned to album.

I feel tags can help resolve this. We could have a selection "No tags" to specifically view media that has not been tagged.

Furthermore, and arguably more important (although I like to tag all my media because I'm OCD), one thing albums lack is viewing more than a single album at the same time when the purpose requires.

For example, I could have an album Cat and an album Dog. It's clear which album to go to when I wish to browser either of them. But if I wanted to view "Cat and Dog" then I need to create a new album titled so and add the media to it.

Tags fix this, in that media tagged Cat, and media tagged Dog could both be browsed at the same time by selecting to view media tagged both Cat and Dog.

This is of course a personal preference. And if Immich has it's heart set on using Albums to conform to more mainstream applications, that is fine. However I hope I can spark a discussion on this, as it is something I would be very eager to see.

(n.b. GitHub requested I select which platform this applies to. It only lets me pick one so I have selected Web but it would apply to the mobile app and server too.)

### Platform

Web

**Comment by bo0tzz** [comment]
The current object recognition functionality already applies tags to images at the moment. The intent is to expand that system in the future so tags are also used for marking eg recognised faces. I think adding support for user-added tags at that point is a logical step 👍

**Comment by palitu** [comment]
I agree (i use Digikam) and it essentially boils down to "logical albums".

The files are all sorted arbitrarily (ie i use YYYY/MM folders), then i use embedded tags within the photo to be able to create albums.  It is the best way of having non-lock-in sorting of photos.  

I would however mark all "auto-generated" tags under some sort of hierarchy (ie `immich-ML/<tags>`).  This will allow us to clean up our photos if there are issues, and be able to distinguish between manual tags and immich tags.

Love your thoughts!

**Comment by nando2301** [comment]
A god idea is to have a "bookmark" icon over thumbnail to see if a photo or video is into album.

Like this:

![image](https://user-images.githubusercontent.com/32605309/200656025-54c6e80e-ca48-4552-907d-5a99dc506d63.png)

Or a Unclassified album por default.

**Comment by AlmightyFrog** [comment]
> The files are all sorted arbitrarily (ie i use YYYY/MM folders), then i use embedded tags within the photo to be able to create albums. It is the best way of having non-lock-in sorting of photos.
I use digikam exactly same way with the addition, that in also separate roughly the source (per Smartphone/camera).

In case Immich would try to support using multi-level tags like digikam does I'd bee more than happy to test and give feedback.
Digikam itself is not bad - it's just like it is not that multi user friendly...
